### PR TITLE
`data` argument to `setup_bootstrap_run` and remove `nm_join` requirement

### PR DESCRIPTION
**Pull request by barrettk**
We use `nm_join()` under the hood to create the _filtered_ starting dataset we resample from. There was previously no method for passing a `.join_col`, so the default `NUM` would always be used under the hood.

The workaround was to add a `NUM` column to the original model and resubmit _before_ attempting to bootstrap it, though this is obviously inconvenient and shouldn't be the behavior. Note that a support ticket has already been filed for this issue, so this PR was initially in response to that.

Rather than providing a new `.join_col` argument, this PR removes the `nm_join()` dependency by parsing the `IGNORE` and `ACCEPT` options in a `$DATA` record, returning a filtered dataset.

Since we had to touch this function anyways, I reached out to @kylebaron about adding a `data` argument as well. There have been [previously discussions](https://github.com/metrumresearchgroup/bbr/issues/678) about this, though mostly offline. I followed the same approach used for `nmsim` models (save new data to the output directory after some checks) with some minor differences.

### CODE REVIEW

**Comment by barrettk** [comment]
@kylebaron requested your review from more of a conceptual and documentation perspective (though feel free to make any additional comments). Im planning on discussing the technical details with @seth127 upon his return for a more formal review

**Comment by barrettk** [comment]
<details closed>
<summary>Local Tests</summary>

```r
> devtools::test(filter = "boot")
ℹ Testing bbr
✔ | F W  S  OK | Context
✔ |        115 | testing bootstrap functionality and running bbi [115.8s]                                                              

══ Results ════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
Duration: 115.8 s

[ FAIL 0 | WARN 0 | SKIP 0 | PASS 115 ]
```
</details>

File: R/nm-file.R [hunk 1]
```diff
@@ -87,20 +88,32 @@ nm_par_tab <- function(.mod) {
...
+  check_model_object(.mod, c(NM_MOD_CLASS, NM_SUM_CLASS, NMSIM_MOD_CLASS, NMBOOT_MOD_CLASS))
   .path <- get_data_path(.mod)
   verbose_msg(glue("Reading data file: {basename(.path)}"))
   .d <- fread(.path, na.strings = ".", verbose = FALSE)
   .d <- remove_dup_cols(.d)
   .d <- as_tibble(.d)
+
   names(.d) <- toupper(names(.d))
+
+  if(isTRUE(filter)){
```
Comment (kylebaron) [inline R/nm-file.R:106]:
```r
Reading data file: acop.csv
  Filtered: 97.5% of records retained
  rows: 779
  cols: 10
```

how about

```r
Reading data file: acop.csv
  rows: 779 of 790
  cols: 10
```

or less like

```r
Reading data file: acop.csv
  rows: 779 (97.5%)
  cols: 10
```

**Review by kylebaron — COMMENTED** [review]
Just one request for less verbose information.

**Review by kylebaron — APPROVED** [review]

File: R/nm-file.R (hunk 1 above)
Comment (barrettk) [reply R/nm-file.R:106]: Went with your first recommendation in [8ad89b9](https://github.com/metrumresearchgroup/bbr/pull/707/commits/8ad89b90acdcb6a6086b68ee82e86b2d5273c975)

**Pull request merged**